core/src/main/codegen/templates/Parser.jj

```diff
@@ -2477,6 +2466,33 @@ SqlNode TableRef2(boolean lateral) :
     }
 }
 
+void TableSample() :
+{
+}
+{
+    (<TABLESAMPLE> | <SAMPLE>)
+}
+
+boolean TableSamplingMethod() :
+{
+    final boolean isBernoulli;
+}
+{
+    // TODO(jsternberg): Only BERNOULLI and SYSTEM should be allowed by default.
+    // Modify the parser to allow additional keywords to be used through freemarker.
+    // Similarly, the empty expression is not part of the default and we should
+    // confirm with the SqlConformance to see if we allow it. Needs to be integrated
+    // properly with upstream calcite.
```

Comment:
Can you create a followup JIRA issue(s), and link it here?

Reply:
I'll link the proposal about refactoring calcite since that's more relevant for this. This todo can't really be applied unless we do that anyway.